akka-docs/src/main/paradox/typed/reliable-delivery.md

```diff
@@ -0,0 +1,358 @@
+warning or deprecation period. It is also not recommended to use this module in production just yet.
+
+@@@
+
+## Module info
+
+To use reliable delivery, add the module to your project:
+
+@@dependency[sbt,Maven,Gradle] {
+  group=com.typesafe.akka
+  artifact=akka-actor-typed_$scala.binary_version$
+  version=$akka.version$
+}
+
+## Introduction
+
+Normal @ref:[message delivery reliability](../general/message-delivery-reliability.md) is at-most once delivery, which
+means that means that messages may be lost. That should be rare, but still possible.
+
+For interactions between some actors that is not acceptable and at-least once delivery or effectively once processing
+is needed. The tools for reliable delivery described here help with implementing that. It can't be achieved
+automatically under the hood without collaboration from the application because confirming when a message has been
+fully processed is business level concern. Only ensuring that it was transferred over the network or delivered to 
+the mailbox of the actor would not be enough, since it may crash right after without being processed.
+
+Lost messages are detected, resent and deduplicated as needed. In addition, it also includes flow control for
+the sending of messages to avoid that a fast producer overwhelms a slower consumer or sends messages at
+a higher rate than what can be transferred over the network. This can be a common problem in interactions between
+actors, resulting in fatal errors like `OutOfMemoryError` because too many messages are queued in the mailboxes
+of the actors. The detection of lost messages and the flow control is driven by the consumer side, which means
+that the producer side will not send faster than the demand requested by the consumer side. The producer side will
+not push resends unless requested by the consumer side.
+
+There are 3 supported patterns, which are described in the following sections:
+
+* @ref:[Point-to-point](#point-to-point)
+* @ref:[Work pulling](#work-pulling)
+* @ref:[Sharding](#sharding)
+
+## Point-to-point
+
+Point-to-point reliable delivery between a single producer actor sending messages and a single consumer actor
+receiving the messages.
+
+Messages are sent from the producer to @apidoc[ProducerController] and via @apidoc[ConsumerController] actors, which
+handle the delivery and confirmation of the processing in the destination consumer actor.
+
+![delivery-p2p-1.png](./images/delivery-p2p-1.png)
+
+The producer actor will start the flow by sending a `ProducerController.Start` message to
+the `ProducerController`.
+
+The `ProducerController` sends `RequestNext` to the producer, which is then allowed to send one
+message to the `ProducerController`. Thereafter the producer will receive a new `RequestNext`
+when it's allowed to send one more message.
+
+The producer and `ProducerController` actors are supposed to be local so that these messages are
```

Comment:
It's checked and enforced.

Reply:
so better if we say they are guanrateed to be local instead of supposed
```suggestion
The producer and `ProducerController` actors are guaranteed to be local so that these messages are
```